compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -549,6 +549,7 @@ class TypeErasure(sourceLanguage: SourceLanguage, semiEraseVCs: Boolean, isConst
    *      - otherwise, if T is a type parameter coming from Java, []Object
    *      - otherwise, Object
    *   - For a term ref p.x, the type <noprefix> # x.
+   *   - For a refined type scala.PolyFunction { def apply[...]: R }, scala.Function0
```

Comment:
This is a tasty-breaking change (if scala 3.0.0 tries to compile a tasty file from a newer compiler, it will crash at erasure since it won't have this case), so this needs a tasty bump and to be part of 3.1.0